crates/driver/src/domain/competition/solution/solver_score.rs

```diff
@@ -0,0 +1,24 @@
+use crate::{boundary, domain::eth, infra::mempool};
```

Comment:
To me, there are two types of scores:

1. `competition::Score` which represents the score to report to the `autopilot` as part of the solver competition. It is just a number.
2. `solution::Score` (or `solution::SolverScore`). This is the score that we get back from the solvers and has a `calculate(self, ...) -> competition::Score` function.

I don't think we need three different score types (and its weird that we need to pass in "mempools" to compute scores, can this be factored out?).

Reply:
Technically, the score cap is part of the competition. We should consider sending it with the `auction` (as it should be configured as a core part of the protocol and not for each driver from a conceptual level).

Reply:
Fixed. https://github.com/cowprotocol/services/pull/1856/commits/7951cd10b758d0239d021ae4577e18070885d841